### PR TITLE
Use ykinfo for checking YubiKey presence in key-script

### PR DESCRIPTION
**Pull request by Vincent43**
* ykinfo doesn't need user interaction and also checks if YubiKey
slot is configured.

* some minor fixes

### CODE REVIEW

**Comment by Vincent43** [comment]
@cornelinux Unfortunately some "minor fixes" were fatal so I had to adjust it. Sorry for that.

I tested current config with both yubikey present at first try and not present and everything seems to work.

File: key-script [hunk 1]
```diff
@@ -19,10 +19,7 @@ message()
     return 0
 }
 
-check_yubikey_present() {
-    ykchalresp -2 AreYouThere > /dev/null 2>%1
-    return $?
-}
+check_yubikey_present="$(ykinfo -q -2)"
```
Comment (cornelinux) [inline key-script]: I like this!

File: key-script [hunk 2]
```diff
@@ -43,7 +40,7 @@ fi
 
 PW="$($cryptkeyscript "$WELCOME_TEXT")"
 
-if check_yubikey_present; then
+if [ "$check_yubikey_present" = 1 ]; then
```
Comment (cornelinux) [inline key-script]: Shoudn't it be 

~~~~bash
if [ "$check_yubikey_present" = "1" ]; then
~~~~
?

File: yubikey-luks-enroll [hunk 3]
```diff
@@ -45,7 +45,7 @@ done
 
 echo "This script will utilize slot $SLOT on drive $DISK.  If this is not what you intended, exit now!"
 
-if [ $CLEAR_SLOT -eq 1 ]; then
+if [ "$CLEAR_SLOT" -eq 1 ]; then
```
Comment (cornelinux) [inline yubikey-luks-enroll]: ``-eq`` is an integer comparison.
So this should be

~~~~bash
if ["$CLEAR_SLOT" = "1" ]; then
~~~~

**Comment by Vincent43** [comment]
@cornelinux  All fixed! I rebased branch on top of master and github now shows some weird but trivial conflict to fix for you :smile: .

**Pull request merged**